trades/accept_trade.js

```diff
@@ -55,6 +60,26 @@ module.exports = ({cancel, id, lnd, secret}, cbk) => {
 
       // Settle the held invoice with the preimage
       settle: ['cancel', ({}, cbk) => settleHodlInvoice({lnd, secret}, cbk)],
+
+      // Open the channel
```

Comment:
You could block settle on open channel success, otherwise if open fails you still took the funds. The disadvantage is you allow free locking of coins, but I guess that's a tricky solve